src/ecMatching.ml

```diff
@@ -695,8 +695,10 @@ let f_match_core opts hyps (ue, ev) f1 f2 =
               Fsubst.f_bind_mem subst hf1.hf_m hf2.hf_m in
           assert (not (Mid.mem hf1.hf_m mxs) && not (Mid.mem hf2.hf_m mxs));
           let mxs = Mid.add hf1.hf_m hf2.hf_m mxs in
+          let lf1 = poe_to_list (hf_po hf1).hsi_inv in
+          let lf2 = poe_to_list (hf_po hf2).hsi_inv in
           List.iter2 (doit env (subst, mxs))
-            [(hf_pr hf1).inv; (hf_po hf1).inv] [(hf_pr hf2).inv; (hf_po hf2).inv]
+            ((hf_pr hf1).inv :: lf1) ((hf_pr hf2).inv :: lf2)
```

Comment:
I think you need to check that the exception are equal. Here you only check that the associated formulas match.